### PR TITLE
Update go dependencies

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind dependency-change

#### What this PR does / why we need it:
Dependabot does not seem to rebase/recreate https://github.com/cri-o/cri-o/pull/8313, so I'm taking over here.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -72,12 +72,12 @@ require (
...
-	k8s.io/api v0.30.2
-	k8s.io/apimachinery v0.30.2
-	k8s.io/client-go v0.30.2
-	k8s.io/cri-api v0.31.0-alpha.0.0.20240529224029-3a66d9d86654
-	k8s.io/klog/v2 v2.130.0
-	k8s.io/kubelet v0.30.2
+	k8s.io/api v0.31.0-alpha.2
+	k8s.io/apimachinery v0.31.0-alpha.2
+	k8s.io/client-go v0.31.0-alpha.2
+	k8s.io/cri-api v0.31.0-alpha.2
```
Comment (sohankunkerkar) [inline go.mod:78]: @saschagrunert Does this version have all the changes specific to https://github.com/kubernetes/kubernetes/pull/117842?
https://github.com/cri-o/cri-o/commit/c931b90a3b4cc3b8b57043c61cca1a05fc977341#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R78

Comment (saschagrunert) [reply go.mod:78]: @sohankunkerkar yes.

**Review by sohankunkerkar — APPROVED** [review]

**Pull request merged**